python/fusion_engine_client/analysis/data_loader.py

```diff
@@ -218,7 +220,7 @@ def generate_index(self, show_progress=False):
                 self.data[MessageType.POSE] = prev_data
 
     def read(self, *args, **kwargs) \
-            -> Dict[MessageType, MessageData]:
+            -> Dict[Optional[MessageType], MessageData]:
```

Comment:
Is this actually a `Union[List[MessageData], Dict[MessageType, MessageData]]` ?

Reply:
Ah I see at the very end you add it to a None entry. I guess I'd find that a bit more confusing then the return signature I'm proposing here since `Dict[Optional[MessageType], MessageData]` could have a mix a `None` and not `None` keys. If you're effectively returning a list, you should probably just return a list.

Reply:
Yes, I originally implemented it that way and returned a list in in-order mode or a dict otherwise. That would be `List[MessagePayload]` though -- a `MessageData` class is a container that holds _all_ of the requested messages. That had consequences for the calling code though when I went to add the ability to return the bytes.

Maybe I'll change to returning the `MessageData` object? That contains two lists: one for payload instances, one for bytes. Basically, return `result[None]` if returning in order.

Reply:
Yeah since MessageData is already just a set of lists that makes sense.